### ISSUE TITLE
Bug? Fails to load a crate file without its data entities  (directories)

### ISSUE THREAD

**Issue by ptsefton** [ResearchObject/ro-crate-py#73]
I think this is a bug.

On loading a directory with ONLY the sample file from here: https://raw.githubusercontent.com/ResearchObject/example-ro-sample-image-crate/main/sample-crate/ro-crate-metadata.jsonld

```
from rocrate.rocrate import ROCrate

crate = ROCrate("sample") 
```

It dies like this:
```
Traceback (most recent call last):
  File "/Users/pt/working/ro-crate-py/test.py", line 3, in <module>
    crate = ROCrate("sample") 
  File "/Users/pt/rocrate/lib/python3.9/site-packages/rocrate/rocrate.py", line 92, in __init__
    self.build_crate(entities, source_path, gen_preview)
  File "/Users/pt/rocrate/lib/python3.9/site-packages/rocrate/rocrate.py", line 226, in build_crate
    raise Exception('Directory not found')
Exception: Directory not found

```

Expected behaviour:

I would expect to be able to load an incomplete crate if I am:
- Writing an HTML renderer
- Writing a validator and I want to report all the errors (rather than dying on the first one

**Comment by simleo** [comment]
Addressed by #75.

What constitutes a bug is mostly a matter of expectations :slightly_smiling_face:

I could not find anything in the spec saying that data entities referenced by the RO-Crate metadata must exist. It's certainly too much to ask of remote items, which could be gone anytime. Actually that's also true of local files, but it's less expensive to check. What matters is what happens when you try to access the referenced resource, and in the current version of the library that only happens when you write the crate to disk. I've adopted a "relaxed" approach: if the files can't be found, they won't be written.

**Comment by simleo** [comment]
In short: with #75, you can not only load, but also write an incomplete crate.
